hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -123,6 +121,25 @@ case class HoodieFileIndex(
     }
   }
 
+  /**
+   * This method traverses StructType recursively to build map of columnName -> StructField
+   * Note : If there is nesting of columns like ["a.b.c.d", "a.b.c.e"]  -> final map will have keys corresponding
+   * only to ["a.b.c.d", "a.b.c.e"] and not for subsets like ["a.b.c", "a.b"]
+   * @param structField
+   * @return map of ( columns names -> StructField )
+   */
+  private def generateNameFieldMap(structField : Either[StructField, StructType]) : Map[String, StructField] = {
+    structField match {
```

Comment:
Please delete the space before ‘:’

private def generateNameFieldMap(structField : Either[StructField, StructType]) : Map[String, StructField] =>

private def generateNameFieldMap(structField: Either[StructField, StructType]): Map[String, StructField]

Reply:
Done, thanks for pointing out code style of scala.